openstack/identity/v2/tenants/results.go

```diff
@@ -69,3 +69,8 @@ func (r tenantResult) Extract() (*Tenant, error) {
 type CreateResult struct {
 	tenantResult
 }
+
+// DeleteResult temporarily contains the response from the Delete call.
+type DeleteResult struct {
+	tenantResult
+}
```

Comment:
I noticed this is also implemented the same way in `v2/users`, but I think this might be incorrect. Delete calls that only return an error and not a resource usually wrap `gophercloud.ErrResult`:

```go
type DeleteResult struct {
  gophercloud.ErrResult
}
```

 This way, you can do the following:

```go
err := tenants.Delete(client, id).ExtractErr()
if err != nil {
  ...
}
```

@jrperritt Do you know of a reason a "raw" result would be returned for `v2/users` Delete?